api/kansas_patron.py

```diff
@@ -0,0 +1,122 @@
+from flask_babel import lazy_gettext as _
+import requests
+import logging
+from authenticator import (
+    BasicAuthenticationProvider,
+    PatronData,
+)
+from config import CannotLoadConfiguration
+from circulation_exceptions import RemoteInitiatedServerError
+from core.model import ExternalIntegration
+from lxml import etree
+
+
+class KansasAuthenticationAPI(BasicAuthenticationProvider):
+
+    NAME = 'Kansas'
+
+    DESCRIPTION = _("""
+        An authentication service for the Kansas State Library.
+        """)
+
+    DISPLAY_NAME = NAME
+
+    SETTINGS = [
+        {
+            "key": ExternalIntegration.URL,
+            "format": "url",
+            "label": _("URL"),
+            "default": "https://ks-kansaslibrary3m.civicplus.com/api/UserDetails",
+            "required": True
+        },
+    ] + BasicAuthenticationProvider.SETTINGS
+
+    log = logging.getLogger("Kansas authentication API")
+
+    def __init__(self, library_id, integration, analytics=None, base_url=None):
+        super(KansasAuthenticationAPI, self).__init__(library_id, integration, analytics)
+        if base_url is None:
+            base_url = integration.url
+        if not base_url:
+            raise CannotLoadConfiguration(
+                "Kansas server url not configured."
+            )
+        self.base_url = base_url
+
+    # Begin implementation of BasicAuthenticationProvider abstract
+    # methods.
+
+    def remote_authenticate(self, username, password):
+        # Create XML doc for request
+        authorization_request = self.create_authorize_request(username, password)
+        # Post request to the server
+        try:
+            response = self.post_request(authorization_request)
+        except requests.exceptions.ConnectionError, e:
+            raise RemoteInitiatedServerError(
+                str(e.message),
+                self.NAME
+            )
+        if response.status_code != 200:
+            msg = "Got unexpected response code %d. Content: %s" % (
+                response.status_code, response.content
+            )
+            raise RemoteInitiatedServerError(msg, self.NAME)
```

Comment:
This is fine, but we also have a module for making requests that could handle some of this: https://github.com/NYPL-Simplified/server_core/blob/master/util/http.py

Reply:
Thanks @aslagle! I didn't realize this existed. I'll see if I can use that instead.